File: JAVA/2558. Take Gifts From the Richest Pile.java
class Solution {
    public long pickGifts(int[] g, int k) {
        for(int i=0;i<k;i++){
            Arrays.sort(g);
            g[g.length-1]=(int)Math.floor(Math.sqrt(g[g.length-1]));
        }
        long s=0;
        for(int i=0;i<g.length;i++){
            s+=g[i];
        }
        return s;
    }
}

File: C++/2558. Take Gifts From the Richest Pile.cpp
class Solution {
public:
    long long pickGifts(vector<int>& g, int k) {
        priority_queue<int>a(g.begin(),g.end());
        for(int i=0;i<k;i++){
            int x=a.top();
            a.pop();
            a.push(sqrt(x));
        }
        long long s=0;
        while(!a.empty()){
            s+=a.top();
            a.pop();
        }
        return s;
    }
};